### PR TITLE
Add connect timeout

### PR DESCRIPTION
**Pull request by ashton-herrington**

### CODE REVIEW

File: connection.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ func (e *ErrUnpack) Unwrap() error {
 // Connection represents an ISO 8583 Connection. Connection may be used
 // by multiple goroutines simultaneously.
 type Connection struct {
-	addr           string
+	Addr           string
```
Comment (adamdecaf) [inline connection.go:49]: Why are we exporting this?

Comment (ashton-herrington) [reply connection.go:49]: This section was a carbon copy of the area from @alovak's PR - can probably un-export, figured I would try to ease his merge conflict =D

**Comment by ashton-herrington** [comment]
See failures, addressing

File: connection.go (hunk 1 above)
Comment (alovak) [reply connection.go:49]: We are using Addr in logs to distinguish connections. Not having access to Addr makes it’s hard

Comment (adamdecaf) [reply connection.go:49]: Ah okay. That's fine, but it wasn't part of "add connection timeout."

**Review by adamdecaf — APPROVED** [review]

**Comment by adamdecaf** [comment]
The CI job is using Go 1.17, so can we merge https://github.com/moov-io/iso8583-connection/pull/15 and then rebase this PR? We should be using Go 1.19 in this project.

**Comment by ashton-herrington** [comment]
@adamdecaf I was going to try to figure this out and ask you about it, but MC ApplePay cert is now unblocked so pausing this for a bit, fine merging and rebasing to me!

**Pull request merged**